webui/cypress/support/step_definitions/base.steps.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { When, Then, Before } from 'cypress-cucumber-preprocessor/steps';
+import { BasePage } from '../../pages/base.po';
+
+const page = new BasePage();
+
+Before(async () => {
+  cy.server();
+  cy.route('GET', '/api/**').as('api');
+  cy.route('GET', '/api/items').as('itemList');
+  cy.route('GET', '/api/items/*').as('itemGet');
+  cy.route('POST', '/api/items').as('itemCreate');
+  cy.route('PUT', '/api/items/**').as('itemUpdate');
+  cy.route('GET', '/api/locations').as('locationList');
+  cy.route('GET', '/api/locations/*').as('locationGet');
+  cy.route('DELETE', '/api/items/**').as('itemDelete');
```

Comment:
what's the difference between `*` and `**`?

Reply:
`**` resolve to any number of levels. `*` resolve to only the next level. So `/api/items/<id>/inventoryTransaction` will match `**` but not `*`.

Reply:
Ahh got it thanks.